App Control v2.0

amogh-samsung opened the pull request:
This is to present an implementation that will wrk for loadable configuration in TizenRT also.

With loadable config, we get the following error with the previous app control structure: -
```
Undefined Symbols in ../build/output/bin/app2
         U pm_ioctl	/root/tizenrt/apps/examples/hello/hello_main.c:81
         U up_mdelay	/root/tizenrt/apps/examples/hello/hello_main.c:94
Makefile.unix:512: recipe for target 'post' failed
```

This is because pm_ioctl cannot be used in loadable scenario even after including in pm.h

amogh-samsung committed:
os: Port SMP functionality from nuttx to TizenRT

Port SMP functionality from nuttx to TizenRT.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

kishore-sn committed:
Create new hello_smp config for sabrelite board.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

kishore-sn committed:
os: Fix SMP runtime issues

When we try to boot up with CONFIG_SMP, there are some runtime
errors and chashes. This commit resolves these issues.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

kishore-sn committed:
os: Use enter/leave_critical section for SMP

TizenRT originally makes use of irqsave and irqrestore API for
to provide critical sections. However, when SMP is implemented, we
need to consider the irq and lock status on multiple cores. This is
implemented in enter/leave_critical_section APIs. When the SMP
configuration is enabled, we need to use these new API's instead of
irqsave/restore.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

jylee9613 committed:
build/tools/amebad/gnu_utility : Update Trustzone image for secure hardfault

* Update Trustzone image that prints secure hardfault message based on security level

CookieDoughMixer committed:
Fix SMP related porting issues and resolve deadlock during bootup
kernel/sched: fix porting error in sched_removereadytorun.c
arm_releasepending.c: uncomment the commented out lines that execute smp code
arm/armv7-a: uncomment all sched_resume_scheduler calls
armv7-a/arm_exit.c: add missing call to sched_resume_scheduler
kernel/sched: revert SMP specific changes for pid hash table
kernel: fix tasklists related macros and attributes add missing entry in g_tasklisttable
pthread.h: fix wrongly assigned pthread default values

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

CookieDoughMixer committed:
examples/smp: Add smp testing example

Added SMP test example from Nuttx to TizenRT.

Modified SMP test app to verify smp opertion using tasks instead
of pthreads and barriers. This is because using pthreads is causing
the system to crash since kernel heap is being corrupted by some
data structures.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

yeetee179 committed:
os/board/rtl8730e: Fix BLE init failure after deinit

yeetee179 committed:
os/board/rtl8720e, rtl8730e: Add BLE server disconnection callback

yeetee179 committed:
os/board/rtl8730e: Add MTU update callback

yeetee179 committed:
os/board/rtl8730e: Fix BLE connection id to start from 24/16 rather than 0

yeetee179 committed:
os/board/rtl8730e: Store BLE write value to feedback to app layer

amogh-samsung committed:
tools/nxfuse: Remove inclusion of security_level.h in case of NXFUSE build

- Fix missing header file security_level.h error during smartfs pre-image build
- security_level.h file need not be copied to Nxfuse sources along with other dependant
 files during NXFUSE_HOST_BUILD as the functionality is not needed during Nxfuse pre-image build.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

kaikaikaime committed:
os/arch/arm/src/amebasmart,os/board/rtl8730e: AmebaSmart I2C support

kaikaikaime committed:
os/arch/arm/src/amebasmart,os/board/rtl8730e: Amebasmart I2S support

jylee9613 committed:
build/configs/rtl8730e : Enable prodconfig

Enable product config in RTL8730E (to check SWD)

jylee9613 committed:
build/configs/rtl8730e : Change default configuration

[Enable configs]
* AUTORESET & BOARDCTL_RESET
* MM_ASSERT_ON_FAIL
* HEAPINFO
* STACK_COLORATION

[Disable configs]
* SYSTEM_READLINE & READLINE_ECHO
* ENABLE_ENV_GET, ENABLE_ENV_SET, ENABLE_ENV_UNSET
* IOTBUS

[Change value]
* MAX_TASKS : 32 -> 128
* NFILE_DESCRIPTORS : 8 -> 64
* PREALLOC_MQ_MSGS : 32 -> 4
* MQ_MAXMSGSIZE : 32 -> 600

ewoodev committed:
build/configs/rtl8720e/loadable_apps: Update configuration

Update rtl8720e/loadable_apps configuration by default.
And change build configuration as below.

- CONFIG_MAX_WDOGPARMS : 2 -> 4
- CONFIG_PREALLOC_WDOGS : 8 -> 32
- CONFIG_WDOG_INTRESERVE : 0 -> 4
- CONFIG_PREALLOC_TIMERS : 2 -> 8
- CONFIG_MAX_TASKS : 64 -> 128
- CONFIG_PRIORITY_INHERITANCE : disable -> enable
- CONFIG_BOARD_INITTHREAD : disable -> enable
- CONFIG_FS_AUTOMOUNT_PROCFS : disable -> enable
- CONFIG_DEBUG_MM_HEAPINFO : disable -> enable
- CONFIG_ELF_CACHE_READ : enable -> disable

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

Commit:
os/board/rtl8730e: fix SoftAP MAC addr
- Set concurrent_enabled to false as concurrent mode is not used
- Fix issue of SoftAP having different MAC address from STA interface

Commit:
os/board/rtl8730e: refine high throughput macro
- Refine high throughput macro definition for CA32, set in platform_autoconf.h

jeongarmy committed:
tools: Move bootparam validation to check_package_size.py

A check_package_size.py validates sizes of all binaries and their partitions.
So move bootparam validation in mkbootparam.py to check_package_size.py.

After modification, We can check bootparam like below.
1) Success
========== Size Verification of built Binaries ==========
Type        Binary Size     Partition Size      used(%)
 KERNEL      1,647,902 bytes    2,117,632 bytes     77.82%    PASS
 APP1            6,367 bytes      512,000 bytes      1.24%    PASS
 APP2          295,315 bytes      512,000 bytes     57.68%    PASS
 BOOTPARAM       8,192 bytes        8,192 bytes     100.0%    PASS
=> Size verification SUCCESS!! The size of all binaries are OK.

2) bootparam partition doesn't exist
FAIL!! No bootparam partition.
It should be located at the end of flash with 8K.

3) bootparam partition isn't at the end of flash
FAIL!! bootparam partition is NOT located at the end of flash.
It should be located at the end of flash with 8K.

4) incorrect partition size with not 8K
FAIL!! Bootparam partition size is NOT 8K.
It should be located at the end of flash with 8K.

yeetee179 committed:
os/board/rtl8730e: fix BLE related SVACE and warnings (#5938)


- add null check for pointers before using the pointers
- replace strcpy with strncpy
- add check for array index to precent array out of bound
- add return to non void function

kaikaikaime committed:
os/arch/arm/src/amebasmart,os/board/rtl8730e: Amebasmart SPI support

kaikaikaime committed:
build\configs\rtl8730e: Add testcase config

kaikaikaime committed:
os/arch/arm/src/amebalite: AmebaLite SPI semaphore fix

ewoodev committed:
os: Remove assertdbg and replace CHECK_SECURE_PERMISSION

we have two methods for controlling the output of assert logs based on security levels: 'assertdbg()'
and 'CHECK_SECURE_PERMISSION()'. However If disable CONFIG_SECURE_DEBUG_ASSERT, It do not affect logs
controlled by 'CHECK_SECURE_PERMISSION()', leading to potential confusion.

To simplify the codebase and align security level checks, we have removed 'assertdbg' and exclusively
rely on 'CHECK_SECURE_PERMISSION()' to manage assert log outputs. But If add it to all assert log outputs,
It is not desirable. So, we collect the output log in units of fault and check it at once.

This change enhances clarity and maintains consistent behavior with security level configurations.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

jeongarmy committed:
build/config/imxrt1020: disable CONFIG_USE_BP

Currently rtk boards only support boot parameter.
So CONFIG_USE_BP should be disabled in this configuration.

Commit:
os/board/rtl8730e: update wificfg for tx power limit
- disable tx power limit to sync wifi settings with tp1x

jylee9613 committed:
build/configs/rtl8730e : Enable SE configs

This commit enables SE configs and NET_SECURITY_TLS for seclink test.
(If disable NET_SECURITY_TLS, it will be failed to build seclink test)
SE configs should be enabled to test Security API and seclink API.

ZhenBei-Sin committed:
os/board/rtl8730e: Fix Peripheral related Build Warning and SVACE

- Deleted the unused API and sources file that caused Build Warning and SVACE
Build Warning
- Added return for the function does return
- Removed the unused variable
- Performed the correct casting conversion
- Solved the shadowed declaration without use the same variable name for the global and local
- Added the define/undefine condition checking for the variable
SVACE
- DREF_OF_NULL.ASSIGN : Performed NULL checking
- SIMILAR_BRANCHES : Removed the idetical branches and keep one only
- NO_CAST.INTEGER_DIVISION & INTEGER_OVERFLOW : Performed the correct casting conversion
- Added the comment for unable solve UNREACHEABLE_CODE

ZhenBei-Sin committed:
os/board/rtl8730e: Fix WiFi related Build Warning and SVACE

 Build Warning
- Added macro checking for declare the unused variable as it depends on the macro on/off
- Changed the NULL checking to 0
 SVACE
- DEREF_OF_NULL.RET.STAT: Added the NULL pointer checking
- BUFFER_OVERFLOW.LIB.EX: Modified the API with correct pass in parameter for the size
- UNREACHABLE_CODE: Added macro to wrap it for cover the switch case break
- OVERFLOW_UNDER_CHECK: Declared the wlan_idx[NET_IF_NUM] with follow thw NET_IF_NUM value
- PROC_USE.VULNERABLE: Replaced strcpy with strncpy with pass in length
- TAINTED_INT.LOOP: Added the condition checking for the unsigned value

yeetee179 committed:
os/board/rtl8730e: Add comment for SVACE fixed array size for itoa

CookieDoughMixer committed:
build/configs/rtl8730e: enable cpp on flat config on AmebaSmart board

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

CookieDoughMixer committed:
rtl8730e/cpp_loadable_8M: enable cpp support on loadable on AmebaSmart

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

kishore-sn committed:
build/configs/rtl8730e : Enable app reloading

Enable app reloading and reload test scenarios.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

kishore-sn committed:
os/board/rtl8730e: Fix heap corruption during app reload

Current implementation reserves some space at the end of BDRAM region
for setting up page tables. This reserved space is getting included at
the end of the kernel heap. However, when we reload the apps, we clear
the page tables completely. As a result the last node information of the
heap gets corrupted. To avoid this, we modify the heap so that it does
not include the areas reserved for page tables.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

kishore-sn committed:
os/arch : Refactor Armv7a architecture code

When we ported armv7a code from nuttx, we have not picked some of the
arch related changes done in tizenrt for armv8m and other architectures.
This commit picks these changes to armv7a.
Following are covered as part of this commit:
- Port tizenrt specific changes from v8m to v7a
- Move duplicate files in v7a to common directory
- Rename some arch files (example arm_internal.h to up_internal.h)
- Refactor and enhance assert code to handle app reload, dumps, etc..
- Add configs to simplify the dumps which will be printed during assert

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

jeongarmy committed:
kernel/binary_manager: Update binary based on partition directly when not using BP

Binary manager operates based on boot parameters.
But if not using BP, if checks their own partitions directly and loads binaries with the latest version.

jeongarmy committed:
apps, loadable_apps: Add missing function call to set boot parameter in binary update samples

For binary update, binary_manager_set_bootparam should be called to set boot parameters.
So I add missing them in binary update samples.

zhongnuo-tang committed:
os/board/rtl8730e: Add RTK Key Derive Function API (PBKDF2)

kishore-sn committed:
build/configs/rtl8730e: Add new config for app L2 page tables

In order to provide memory protection for apps using MMU, we need to
configure page tables for each app. Each app can have 1 L1 page table
and many L2 page tables. During boot up, we reserve memory space to hold
all the page tables required by apps. So, we need to know how many L2
page tables are required for each app. This commit adds a config variable
to define the number of L2 page tables per app.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

Commit:
os/drivers/compression: Adds compression driver module
	- adds compression api at public header(compression.h)
	- adds compression module in driver , so that it can be used by application
	- driver is able to return the compressor compression type

Commit:
testcase/le_tc/compression: Adds compression driver apis testcase

	adds testcases that test compression, decompression of sample
	unsigned char data(through ioctl calls to compression driver)

kishore-sn committed:
apps/examples, build/configs, os/drivers: Rename MPU test

Rename MPU test to MEM_PROTECT test, because memory protection
can be provided either by MPU or MMU.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

Commit:
os/board/rtl8730e: Modify FTL Flash access in NP to avoid error

-Add FTL IPC in NP to remove flash access in NP
-Add FTL IPC in AP to receive and operate flash in AP

kaikaikaime committed:
os/arch/arm/src/ameba: GPIO pull up/down fix
The deinit irq function also clears out the previous parameters set.

jeongarmy committed:
board/rtl8730e: Add choice to select flash size of board in Kconfig

There is 2 kinds of flash capacity of rtl8730e board: 8Mbyte, 16Mbyte.
The related config values should be set according to flash size user selected.

jeongarmy committed:
build/configs/rtl8730e: Add config for board flash size

Update defconfigs according to adding configurations about flash size.

Commit:
os/arch/arm/src/armv7-a: Add missing initialization of adj_stack_ptr

	this commit initializes tcb->adj_stack_ptr in up_create_task func
        in arm_createtask.c, which was not being initialized before

Commit:
build/configs/rtl8730e: Enable memory leak checker configs

	Enabled required configs of memory leak checker in every configuration's defconfig

Commit:
arch/arm/include/amebasmart: Adds memory_region.h for rtl8730e for mem_leak test

	- Adds memory_region.h file for rtl8730e to perform memory leak checker test

	  Following are the results :

	  TASH>>mem_leak

	  Kernel :
	  *** NO MEMORY LEAK.

	  Below are text addresses of loadable apps (and common binary if enabled) :
	  The pc value of the allocation can be obtained by subtracting the text start address of the appropriate binary

	  [app1] Text Addr : 0x6038c000, Text Size : 8192
	  [app2] Text Addr : 0x60410000, Text Size : 294912

	  app1 :
	  *** NO MEMORY LEAK.
	  app2 :
	  *** NO MEMORY LEAK

thapav committed:
tools/trap: Update TRAP tool as per latest assert logs

This patch updates the parser as per latest assert logs in arch files
removing the dependency of assert logs on any fucntion API's.

vidisha@vidisha:~/rt/tools/trap(master)$ python3 ramdumpParser.py -t log

*************************************************************
dump_file                   : None
log_file                    : log
Number of binary            : 1 [kernel]
"kernel" elf_file           : ../../build/output/bin/tinyara.axf
*************************************************************

DEBUG_MM_HEAPINFO is not enable. Enable DEBUG_MM_HEAPINFO to see heap usage
-----------------------------------------------------------------------------------------
1. Crash Binary             : kernel

2. Crash type               : code assertion by code ASSERT or PANIC

3. Crash point
	-  print_assert_detail: Assertion failed at file:hello_main.c line: 71 task: appmain

4. Code asserted in:

	- Code asserted in normal thread.

5. Call stack of last run thread

Stack_address	 Symbol_address	 Symbol location  Symbol_name		File_name
 User stack
0x1006f8	 0xe003d9c	 kernel  	  vfprintf	/root/tizenrt/os/include/stdio.h:479
0x100700	 0xe003dd2	 kernel  	  stdoutstream_flush	/root/tizenrt/lib/libc/stdio/lib_stdoutstream.c:100
0x1006e20c	 0xe08a9ed	 kernel  	  __FUNCTION__.5309
0x1006e23c	 0xe006bcc	 kernel  	  hello_main	/root/tizenrt/apps/examples/hello/hello_main.c:67
0x1006e240	 0xe001afc	 kernel  	  task_start	/root/tizenrt/os/kernel/task/task_start.c:133

h. Heap Region information:

	!!!! NO HEAP CORRUPTION DETECTED !!!!

Checking kernel heap for corruption
Checking corruption (0x1005b260 - 0x10076ff8) : Heap NOT corrupted
Checking corruption (02002130 - 0x023bfff8) : Heap NOT corrupted
No Kernel heap corruption detected.

x. Miscellaneous information:

List of all tasks in the system:

                          NAME |   PID |  PRI | TOTAL STACK | STACK ALLOC ADDR | TCB ADDR
 -----------------------------------------------------------------------------------------------
                       Idle Task |     0 |    0 |    1024 |       0x10004bfc | 0x1002bd80
                          hpwork |     1 |  224 |  2028 |       0x1005dcf0 | 0x1005d690
                          lpwork |     2 |   50 |    2028 |       1005eb60 | 0x1005e500
                     LWIPCP/IP |     3 |  105 |    4068 |       0x100660f0 | 0x10065a90
            netmgr_event_handler |     4  100 |    2012 |       0x10067760 | 0x10067100
                           tash |     6 |  125 |    4076 |       0x10069470 | 0x10068e10
               ifi msg handler |     7 |  100 |    4060 |       0x1006af90 | 1006a9d0
                 ble msg handler |     8 |  100 |    4068 |       0x1006c600 | 0x1006bfa0
                         appmain |     9 |  100 |    28 |       0x1006da60 | 0x10068900
----------------------------------------------------------------------------------------------------

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

jeongarmy committed:
build/configs/rtl8730e: Increase 256KB size of 'reserved' partition for 16MB flash

Increase 256Kbyte size of 'reserved' partition for additional usages for 16MB flash.

jeongarmy committed:
build/configs/rtl8730e: Rename 'tc' directory to 'tc_8M'

Rename 'tc' directory to 'tc_8M' for better understanding.

zhongnuo-tang committed:
build/tools/amebasmart/bootloader: print Flash size

1. km4_bootloader_ver_47469fd
2. "flash_size: 8M/16M" will be printed accordingly

jeongarmy committed:
build/configs/rtl8730e: Resize binary partitions for loadable apps

Resize binary partitions for kernel and apps

kishore-sn committed:
os, apps: SMP code refactoring

- Fix build errors in smp and non smp projects
- Fix coding issues and review comments

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

Commit:
build/tools/amebasmart: Modify Coex Case of Wifi Scan

-Use PTA instead of TDMA when doing Wifi Scan

amogh-samsung committed:
Update RTL8730E defconfig files

- Update default configurations for RTL8730E
- Enable/disable configurations as per latest use and requirement.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
circleci: Update test configurations for RTL8730E

- Names of defconfigs 'hello_8M' and 'loadable_apps_8M' are updated to 'hello'
 and 'loadable_apps', respecitvely.
- Modify circleci test script accordingly to use these new defconfigs.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
Update Kconfig files w.r.t. Rtl8730e

- Update Kconfig files to make selections based on Rtl8730e configs.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

zhongnuo-tang committed:
build/tools/amebasmart/gnu_utility/km0_km4_app: fix boot up mixed log

1. fix mixed log due to different cores during boot up

zhongnuo-tang committed:
build/tools/amebasmart/bootloader: Boot up log baudrate set according to eFuse value

1. bootloader version b5602c7_2023/10/19-17:42:15
2. print current LogUart baudrate according to eFuse "LogUart Baudrate: 115200/1500000"
3. bootloader log and hotfix log baudrate will depend on eFuse

gexuyan committed:
os/board/rtl8721csm: fix assert with wifi scan with ssid (#5963)

* In wifi driver, wifi scan process will exit directly when the driver fw state is "under scan" or "under wifi connection". Before this commit the "len" param in scan buf is not cleared in this case. It will make parse_scan_with_ssid_res() runs into app_scan_result_handler() many times with empty scan_result_report and cause the heap corruption issue. This commit reset scan buf to all 0 in wlan driver in this case to fix the assert issue.

* previously when wpa3 sae auth fail, driver didn't change the fw state from "under wifi connection" to "not under wifi connection" immediately. This will lead to wifi scan is exited even though wifi connection already failed and completed. This commit adds reporting join status after wpa3 sae auth fail to change the driver fw state immediately in this case.

Co-authored-by: gexuyan <gexuyan@realtek-sg.com>

ZhenBei-Sin committed:
build/tools/amebasmart: Add ATF layer Fault Handler

- Updated the secure layer bin, dump and map file
- Removed the bl2_bl32.bin as no longer requires [PR#5918]

ewoodev committed:
os: Change CHECK_SECURE_PERMITSSION() to !IS_SECURE_STATE()

Change CHECK_SECURE_PERMISSION macro to IS_SECURE_STATE macro.
The CHECK_SECURE_PERMISSION macro checks availble to access or to print secure infomation,
but IS_SECURE_STATE macro checks if now OS state is secure state that should protect secure infomation.

The reason for modifying this is CHECK_SECURE_PERMISSION having some minor problem as below.
1. CHECK_SECURE_PERMISSION means checking secure permissions, but we need to check what the return value is.
2. get_security_level() == LOW_SECURITY_LEVEL is increases the operation slightly.
3. When CONFIG_SECURITY_LEVEL is disabled, It returns (1). but it make confuse, becasue This valse is not a HIGH_SECURITY_LEVEL(1).

This change fixes the minor issue above and improves readability.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

ewoodev committed:
mfdbg, heap_dbg: add security_level on user mode

In loadable builds, the user heap and kernel heap operate completely separately.
Only in some cases, such as when an allocation failure occurs during operation in the User Heap,
does the code execute by changing to the kernel area (privileged mode).

However, currently in mm modual, security_level is applied only in kernel area (privileged mode).
Since mm log is designed to output physical memory, it can cause potential security issues,
so if the OS is secure level state, security_level must be applied to not output memory in the user area as well.

Therfore, Change mfdbg and heap_dbg, which output mm logs, to check security_level in all env.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

hk-gwak committed:
framework/ble_manager_msghandler.c: prevent to break sem by interrupt

Prevent to break sem by interrupt.

Co-authored-by: j11591-kim <j11591.kim@samsung.com>

ewoodev committed:
os/arch/arm: Apply reboot reason in armv7-a

This commit adds two as below.
1. store reboot reasons that occur in armv7-a/ codes
2. print stored reboot reason during booting on armv7-a boards.

And, To use reboot_reason_write_user_intended in up_assert(),
add up_reboot_reason.c of the amebasmart chip that uses armv7-m to be built.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

CookieDoughMixer committed:
arch/arm/amebasmart: add missing SPI slave select code for RTL8730E

Add missing SPI slave select code, without which we get wrong results

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

ewoodev committed:
build/configs/rtl8730e: enable CONFIG_SYSTEM_REBOOT_REASON

Enable CONFIG_SYSTEM_REBOOT_REASON for the rtl8730e board.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

zhongnuo-tang committed:
os/arch/arm/src/amebasmart/: implement reboot reason code

1. support reboot reason code same as TP1X and TP1E.

sahil1dotgupta committed:
lib/libc/math: Update libc/math from nuttx commit

os/include/tinyara/math.h: Update libc/math from nuttx commit to fix libc.
lib/libc/math: Add new files for function definations corresponding to function declarations added in os/include/tinyara/math.h.

Issue fixed: Libc caused build issues in libcxx.
libcxx/math.h functions' definitions are not found due to issue in old libc/math.h, which has definitions of same functions in form of macros.

This commit by Nuttx changes the macros to function declarations and defines them in separate files.
Nuttx commit id: https://github.com/apache/nuttx/commit/df812d09a219495a8ba1d0c30a2e9ebc05c37f3b

Commit message by nuttx:

math: Make this friendly with libcxx
- Turn some macros into functions

- Implement some type-agnostic functions.
  (Just use __builtin_xxx)

jeongarmy committed:
os/tools: Check partition configurations when build

It is necessary to check if partition configurations are set well when build to avoid errors that occurs in run-time.

- Success
========== Verification of partition settings ==========
=> Partition verification SUCCESS!! The setting of all partitions is OK.

- Failure
Ex1) Set greater size than flash size
========== Verification of partition settings ==========
!!!!!!!! ERROR !!!!!!!
The configured partition size is larger than board flash size. Please Re-configure.
The size of configured size : 8389632, Board flash size : 8388608

Ex2) Missed some entries
========== Verification of partition settings ==========
!!!!!!!! ERROR !!!!!!!
The number of entries of partition list are mismatched. Please Re-configure.
CONFIG_FLASH_PART_SIZE (13) : "16,8,16,472,8,1792,1792,512,512,1016,1016,1024,8,"
CONFIG_FLASH_PART_TYPE (12) : "none,none,none,none,none,kernel,kernel,bin,bin,bin,smartfs,bootparam,"
CONFIG_FLASH_PART_NAME (13) : "bl1,bl2,reserved,ss,system_data,kernel,kernel,app1,app1,app2,app2,userfs,bootparam,"

jeongarmy committed:
build/configs/imxrt1020-evk: Delete wrong entry in partition size list

Delete wrong entry in partition size list in imxrt1020-evk/loadable_elf_apps configuration.

jeongarmy committed:
os/tools: Add missing global keyword for global variable usage

'global' keyword should be added for global variable usage in the function.

CookieDoughMixer committed:
os/dutils.sh: fix path to crash logs in when calling trap tool

The path to the crash logs when using trap with dutils is taken relative
to the os folder (ex : ../tools/trap/testlogs) but the same path is also
used when calling trap command,

dutils.sh log : Executing: ramdumpParser.py -t ../tools/trap/testlogs -e tinyara.axf

which results in the following error,

../tools/trap/testlogs does not exist. Please provide the proper path for log_file...

So, added comment to provide the path relative to os folder and also have
corrected the crash logs path by adding /root/tizenrt/os prefix

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

sahil1dotgupta committed:
lib/libc/math: Add Samsung 2023 license

lib/libc/math: Add license to new libc files.

sahil1dotgupta committed:
lib/libc/math: Format newly added libc files

lib/libc/math: Format as per TizenRT coding rule.

kaikaikaime committed:
os/board/rtl8721csm: Remove GPIO PA12 initialisation

Current GPIO PA12 is in conflict with UART1 init

zhongnuo-tang committed:
build/tools/amebasmart/bootloader: implement RDP verification feature

1. add log for secure boot status, eg: "IMG1 SBOOT OFF/EN, IMG1 SBOOT SUCCESS"
2. add log for RDP status, eg: "RDP EN/OFF, RDP SUCCESS/FAIL"
3. implement feature to verify RDP, if RDP is encrypted with wrong key/iv,
   before-> module will be stuck during booting.
   now-> "RDP FAIL" will be printed.
4. please use secure_boot_script_amebasmart_v03_official to sign RDP/Secureboot.
5. increase hotfix wait time from 50ms to 100ms, please change BOOTLOADER_STRING='Autoboot in 100 milliseconds' in hotfix script
6. remove some unnecessary logs to prevent mix log

aadotverma committed:
framework/lw_aifw, framework/aifw: Update AI Framework

framework/lw_aifw: Remove old Light Weight AI Framework
framework/aifw: Add AI Framework version 2.0

aadotverma committed:
os/Kconfig: Update AI Framework configs

aadotverma committed:
apps/examples: Update AI Framework sample application

apps/examples/lw_aifw_sine_test_app: Remove sample application showcasing usage of old AIFW.
apps/examples/aifw_test: Add sample application showcasing usage of AIFW 2.0

Commit:
drivers/compression: fix typos

	This commit fixes the typos in os/drivers/compression/README.md file
	and os/compression/compress.c file.
	succesful -> successful
	funtion -> function

Commit:
os/board/rtl8730e: Change alignment of malloc
- Use kmm_memalign instead of kmm_malloc in RTK malloc operations to align malloc to 64b to resolve cache operation issues on CA32
- Fix DCache_Clean for ssid array during wifi scan to ensure cache is cleaned properly for cache coherency

aadotverma committed:
framework/aifw, examples/aifw_test: Resolve review comments

1. framework/src/aifw/Kconfig: Resolve warnings
2. framework/src/aifw/TFLM.cpp: Change data type of variable storing
output of ftell() from int to long.
3. framework/aifw: Replace #ifndef include guards with #pragma once
4. examples/aifw_test: Apply TizenRT coding style

aadotverma committed:
examples/aifw_test: Remove AIDataBuffer NULL check from process handler

Commit:
build/tools/amebasmart/gnu_utility: Fix Secure Storage 'DATA_IN_USE' issue

amogh-samsung committed:
build/configs/rtl8730e: Enable neccessary configs for jSmartFS

- Enable CONFIG_MTD_BYTE_WRITE which helps in journal update efficiency
- Enable 16-bit CRC and Journaling for jSmartFS

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

zhongnuo-tang committed:
add build/tools/amebasmart/gnu_utility: enable WIFICSI

1. enable WIFICSI driver code callback

ewoodev committed:
armv7-a/arm_assert: fix board reset in assertion

If CONFIG_BOARD_ASSERT_AUTORESET is enabled and occur crash(assertion), board will be rebooted.
But, rebooting does not work with armv7-a, because CONFIG_BOARD_ASSERT_AUTORESET is not applied in armv7-a/arm_assert.c yet.

Therefore, Fix so that it reboots correctly when an assertion occurs.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

Commit:
os/board/rtl8730e: Added multi I2C support
- Added multi I2C support
- Fixed GPIO and Audio shared pin configuration

amogh-samsung committed:
os/fs/smartfs: Skip smartfs_sector_recovery in Nxfuse image build

- smartfs_sector_recovery() need not be invoked during Nxfuse image build because
 we are building a fresh image to flash to the partition and hence recovery is not
 required.
- Additionally, in the case of Nxfuse host build, we will always get the error message
 "CRITICAL BUG!! Root sector is gone!!" from smartfs_sector_recovery despite successful
 completion.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

Commit:
os/compression/compress_read.c: Fixes memfault issues during booting

	- writesize varibale not assigned before passing to the function
	  decompress_block, which will cause memfault error

amogh-samsung committed:
build/configs/rtl8730e: Update specific RTL8730E defconfigs as required

- Disable CONFIG_DISABLE_OS_API
- Update CONFIG_MAX_WDOGPARMS=2 -> 4, CONFIG_PREALLOC_WDOGS=8 -> 32
 CONFIG_WDOG_INTRESERVE=0 -> 4 and CONFIG_PREALLOC_TIMERS=2 -> 8
- Enable CONFIG_LOG_DUMP
- Update CONFIG_SCHED_LPWORKSTACKSIZE=2048 -> 4096
- Disable CONFIG_SYSTEM_IPERF
- Disable CONFIG_ENABLE_KILL
- Disable CONFIG_ENABLE_KILLALL
- Disable CONFIG_SCHED_CPULOAD and CONFIG_ENABLE_CPULOAD

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

ewoodev committed:
os/kernel/log_dump: fix device hang by kernel heap locked

Context switch is possible when between after check kerenl heap locked and before send mq for wake up log_dump thread.
If in this time, context switch and other thread get kernel heap sem, log_demp thread wake up with kerenl heap locked.
This can occur device hang.

Therefore, Fix to do not context swiching during kernel heap locked checking and mq sending to log_dump thread.
And, Change meaningless spin-locking in the while statement to sleep every 10ms.

Additional, detail of possible device hang as belog.
  - Thread 1: take uart sem and wait for complate compress from log_dump thread.
  - Thread 2: take kerenl heap sem and wait for take uart sem
  - Thread log_dump: wait kernel heap sem.
    -> Circular wait, dead lock.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

aadotverma committed:
framework/aifw: Sync AI Framework

1. Add brief description of class, namespace, and structure.
2. Add API to clear sensor data w.r.t model.
3. Add better logs for data debugging.

Commit:
os/board/rtl8730e: peripheral and soc folder sync for 8.4d sdk

* update peripheral and soc folder to 8.4d release sdk
* update pinmux config function

ewoodev committed:
armv8-m/up_schedulesigaction: Fix mem_fault caused by not setting REG_EXC_RETURN

On loadable build, If the interrupt handler changes the pc to up_sigdeliver for a running tcb in up_schedule_sigaction,
a memory fault is occured when the running thread tries to perform up_sigdeliver() after interrupt handler ends.
This memory fault because the running tcb is not changed to privileged mode, so can't access up_sigdeliver, a kernel resource.
Because running thread is un-privileged mode, but current_regs[REG_EXC_RETURN] is not setted EXC_RETURN_PRIVTHR in up_schedule_sigaction.

After interrupt handler is terminated, we copy current_regs[] to h/w register,
and check r14 to determin whether to run thread in privileged or un-privileged mode in up_exception.S

But, EXC_RETURN is automatically populated into the Link Register(LR) on entry to an exception.
In other words, during exception, r14 is EXC_RETURN, not LR.

Therefore, when up_schedule_sigaction is called by interrupt handler,
current_regs[REG_EXC_RETURN] must be setted EXC_RETURN_PRIVTHR for run up_sigdeliver with privileged mode.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

CookieDoughMixer committed:
Revert "os/binfmt: Modify caching method for symtab and reltab"

This reverts commit 5ae1088fbf0250eda3e8bf18907707b1ded9d659. ELF caching
doesnt not work with this commit applied. Hence reverthing this commit. Will
recheck and apply it in a later patch

jwei5 committed:
os/board/rtl8721csm: add feature for selecting wpa mode (#5987)

- Add feature to select the wpa mode to use to connect to APs in mixed security mode based on security type indicated by user

Co-authored-by: jw <jw@ubuntu.com>

CookieDoughMixer committed:
configs/rtl8721csm/loadable_apps: enable elf cache configs

elf cache configs got disabled because of the previous revert
commit. So, enabling them in this commit

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

jeongarmy committed:
tools/check_pacakage_size.py: Declare 'global' keyword once

In python, 'global' keyword is needed for usage of global variable.
It is enough to declare 'global' keyword once in a function because 'if' context doesn't make new scope.
With this change, the below build warning is fixed.

========== Verification of partition settings ==========
=> Partition verification SUCCESS!! The setting of all partitions is OK.
./tools/../tools/check_package_size.py:72: SyntaxWarning: name 'FAIL_TO_BUILD' is assigned to before global declaration
  global FAIL_TO_BUILD

aadotverma committed:
external/tfmicro: Add operators with CMSIS_NN optimization

kaikaikaime committed:
os/arch/src/amebasmart: Add multiple SPI support

Add menuconfig selection for multiple SPI
Change pins and roles

Commit:
build/tools/amebasmart/gnu_utility: Add trustzone files for 8.4d

Commit:
os/board/8730e: add wifi files from 8.4d porting

* build/configs/rtl8730e/scripts: Sync .ld changes from 8.4d
* build/tools/amebasmart/gnu_utility: sync KM4 image to 8.4d, add release note for libs
* os/arch/arm/src/amebasmart: update Make.defs

zhongnuo-tang committed:
build/tool/amebasmart/bootloader: sync bootloader for v8.4d

1. add fault handler
2. fix some typo
3. update flash layout

Commit:
os/board/rtl8730e: Update KM4 image, add buffer print feature

* buffer print feature is disabled by default
* Only certain logs that cause mix logs will use buffer print as some logs may be stuck in inic_ipc_api_dev_message_send
* Update KM4 image with customizations for powersave

sahil1dotgupta committed:
os/include/tinyara/math.h: Remove commented code.

os/include/tinyara/math.h: Remove old macro defining signbit(). As signbit() function was defined by a new macro in
commit: https://github.com/Samsung/TizenRT/commit/613836a32a790f4c80400407701fd7baf9cdec41

kaikaikaime committed:
build/configs/rtl8730e: Enable SPI0/1 in defconfig

Taejun-Kwon committed:
media/am : Continue to find audio_out card if it fails find audio_in Most of card support only one type of card, hence it should continue to find other card

ewoodev committed:
os/arch/arm/amebasmart: Change peripheral config to board level config.

amebasmart_i2c.c have dependency CONFIG_AMEBASMART_I2C_XXXX configuration
So, when deciding whether to include this file in build, we should check CONFIG_AMEBASMART_I2C, not CONFIG_I2C.
The same goes for amebasmart_spi.c.

Therefore, Change CONFIG_I2C and CONFIG_SPI to each CONFIG_AMEBASMART_I2C and CONFIG_AMEBASMART_SPI.

Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

yeetee179 committed:
os/board/rtl8730e: change BLE to single core on AP

yeetee179 committed:
os/board/rtl8721csm: fix fault when flash write and BLE pairing happens together

yeetee179 committed:
os/board/rtl8730e: fix 8.4d porting related issues and warnings

*increase max queue number for BLE indication queue
*fix BLE related warnings
*add .gitignore file to prevent missing
*move header files and shorten CFLAGS string
*add missing wifi security types
*fix mixed log issue
*move km4 version to ap layer
*Fix OCF Issue
*change to use zmalloc instead of malloc in ipc api

edwakuwaku commented on apps/examples/hello/hello_main.c:
```diff
@@ -71,43 +76,88 @@ int hello_main(int argc, char *argv[])
-	 */
-	// pm_ioctl(PM_IOC_STAY);
+
+	int fd_lock, fd_unlock, fd_sleep;
```
Since the application layer will be controlling with the help of pm_procfs.c, means that we don't need pm_ioctl.c anymore rite? If then, I will move out the logic and integrate with pm_procfs.c, also provide the rest which is necessary.

edwakuwaku replied:
After thinking, I think we still have to keep pm_ioctl.c, the power_xxx_write api will be the middle layer to reach pm_ioctl, since the application should not have any knowledge about PM state, and also for easier maintenance in the future. Please share your thoughts on this~

amogh-samsung replied:
1. Yes, the functionality of pm_ioctl.c has to be included in pm_procfs.c (preferably, through the existing APIs available).
2. "ioctl" is not available as a syscall for procfs_operations (https://github.com/Samsung/TizenRT/blob/0f25fb2c02dc4964019bb736ca979af2e03306e9/os/include/tinyara/fs/procfs.h#L70 ), so we cannot use ioctl directly in our applications.
If you want that some procfs syscall gets transformed into a pm_ioctl call, we may be able to do that.
Could you clarify why we should do that AND/OR share some basic implementation of your idea please?

edwakuwaku replied:
Hi @amogh-samsung ,

2) Hmm, but my meaning here is to invoke pm_ioctl in the pm_procfs, and application will invoke syscalls as it is (which is present in the structure here you have provided).

amogh-samsung replied:
Okay, so you mean?
App -> pm_procfs.c -> pm_ioctl.c -> pm_stay/relax

If yes, why do you prefer this control flow?
The same requirement can be handled by write(fd, PM_NORMAL, 1)
Where fd is an open file descriptor for /proc/power/domains/0/power_lock OR power_unlock
Is there some advantage in ioctl that I am missing?

edwakuwaku replied:
> After thinking, I think we still have to keep pm_ioctl.c, the power_xxx_write api will be the middle layer to reach pm_ioctl, since the application should not have any knowledge about PM state, and also for easier maintenance in the future. Please share your thoughts on this~

My opinion is over here, if you think that it is redundant, I can also integrate to just work in pm_procfs.c~

amogh-samsung replied:
Ok, lets avoid the use of the intermediate ioctl.
As long as we can get the work done with pm_procfs.c

Tirtha-Samsung committed:
os_bringup.c: Increase the priority of preapp_main

  preapp_main should run before user_main

  increase priority of preapp_main to default + 1

Signed-off-by: biswas-t <biswas.t@samsung.com>

kaikaikaime committed:
os/board/rtl8730e/src/component: Audio driver update

kaikaikaime committed:
os/arch/arm/src/amebasmart: I2S driver & port update

* 2 I2S Adaptor setup
* TX has been verified

CookieDoughMixer committed:
Add support for ACL1019 AMP device

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

Commit:
os/board/rtl8730e: fix build break
* remove ameba_ftl.c from build
* add ameba_system.c into build

CookieDoughMixer committed:
audio/resample: correct the calculation of input buffer size

the number of frames in the intermediate buffer for resampling
are calculated on the old number of channels, which is wrong.
So, calculate it based on the new number of channels.

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

yeetee179 committed:
os/board/rtl8730e: enable CONFIG_FTL_ENABLED in all of defconfigs

ZhenBei-Sin committed:
build/tools/amebasmart: Fix Secure FTL Init Issue


Found the possibility to stuck working during boot :
- During Secure FTL init, mbedtls init & flash init process have to need for FTL load / storage action
- If Secure FTL init faster than SE init, It can be possible to stuck the boot process.

Modification : Added the mbedtls init & flash init at Secure FTL init step

yeetee179 committed:
os/board/rtl8730e: Add CONFIG_AMEBASMART_FTL in Kconfig and defconfig

Commit:
os/board/rtl8730e: Set alignment requirements in rtl8730e malloc

- Set malloc operations used by rtl8730e source code to be in size multiples of 64 bytes
- This means if malloc used by rtl8730e source code is 30 bytes, actual malloc will be 64 bytes, but most rtl8730e operations will not retain heap for a long time and will be freed
- Malloc used by rtl8730e source code that involves cache operations have to be 64 bytes aligned and also in multiples of 64 bytes, as CA32 cache line is 64 bytes
- This is to ensure cache operations do not overwrite into heap, which can cause heap corruption and fault issues
- Change global variable "align" used in osdep_service_memory to use preprocessor #define to protect variable from being overwritten

jiangxiaoyunjoy committed:
build/tools/amebasmart/image_tool: Add flash size check and flash protection feature
Add flash size checking process before downloading images :
- If the image size exceed module flash size, report error and stop download
Add flash protection feature during the download process:
- Check flash status before erasing process, unlock the flash if the flash status is locked

jylee9613 committed:
framework/include/security : Increase secure storage data slot size

There is a slot that stores AP information, but the required size has increased while storing multi-AP information.
Since the number of slots is limited, it is more efficient to store the data by increasing the size of existing slots instead of dividing it into several slots.
Therefore, this commit increase data slot size from 256 to 4096.
SECURITY_MAX_CERT_BUF is also changed to 4096 for the identity with trt_se.h (product code)

ZhenBei-Sin committed:
build/configs/rtl8730e: Move CA32 to XIP and KM4 remain at PSRAM

Changes Note:
CA32 ATF 1.5M & KM4 PSRAM 1387KB.
 * The CA32_BL2_DRAM_NS end address and available heap need to further check and modify
1. Move KM4 code to PSRAM
2. Keep 5secs delay booting issue temporarily
3. Undefined the CONFIG_ARM)HAVE_MPCORE as Cortex-A32 does not require
4. Modify the defconfig for the RAM size
5. Adjust the arm_head.S to setup the CP15_SCTLR register flow

edwakuwaku committed:
os/arch/arm/src/amebasmart, os/board/rtl8730e: Minor Cleanup for RTL8730E (#6025)

- irq definition update
- initial bringup flow cleanup
- some log fixes
- remove unecessary api
- update array size to tally with total SPI number

amogh-samsung committed:
build/configs/rtl8730e: Align defconfigs with generated menuconfig outputs

- Remove unnecessary options from defconfigs which are not present when generated using menuconfig option

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

Wuth5 committed:
os/board/rtl8730e/src/component: fix wifi connection failure with wpa23_aes mix mode

edwakuwaku committed:
os/arch/arm/src/armv7-a: Place arm_mmu.c to PSRAM

zhongnuo-tang committed:
os/board/rtl8730e/src: extend watchdog timeout (#6030)

1. watchdog Timeout data type change from u16 to u32
2. adjust prescaler so that maximum timeout value increase from ~65s to ~188s

ZhenBei-Sin committed:
build/configs/rtl8730e: Reduce KM4 PSRAM and CA32 ATF size

Changes Note:
Total free up 6MB
- KM4_BD_DRAM reduce from 1388KB to 1MB
- CA32 ATF reduce from 1.5MB to 1MB
- Remove the KM4_AUDIO_BUFFER 128KB (SRAM) is for VAD as no required to use

ewoodev committed:
kernel/prctl: Add PR_GET_TGTASK cmd for group ID

It support to get the main task ID in own group.
If parent-child relationship is not supported, the ID of its own task/thread is returned.

Signed-off-by: ahreum-jeong <ahreum.jeong@samsung.com>
Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

ZhenBei-Sin committed:
os/board/rtl8730e: Move Flash Related API to PSRAM

Issue:
When perform smartfs "format" will causing hang.

Root Cause:
After implement the XIP changes, the flash api was default located to the XIP.
For the flash related API must be locate to PSRAM.

Changes Note:
Add the SRAMDRAM_ONLY_TEXT_SECTION for the flash related API move to PSRAM.
Add the FLASH_Write_Lock/Unlock to make sure before perform the flash operation, CA32 must be disable the IRQ.

CookieDoughMixer committed:
mm/mm_heap: initialize mm_delaylist element to NULL during mm_initialize

mm_delaylist nodes are freeded during before a new malloc. If mm_delaylist
has stale value, this will result undefined behaviour (mostly data abort as
we try to access random address). Hence, initialize it to NULL.

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

zhongnuo-tang committed:
build/tools/amebasmart/gnu_utility: fix User Image Verification Issue

1. the root cause of this user image verification issue is there is a mismatch in data type between
Secure API and TizenRT Non Secure entry API
2. fix by ensure data type are matched

CookieDoughMixer committed:
kernel/task : retain task_state in tcb until task_exithook is finished

Clean up in the above exithook function is dependent upon the task
state during the exit (task can be blocked on mq and a differnt task
could have killed it). As a result, we need to retain the task_state
in tcb until exithook is done

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

amogh-samsung committed:
apps/examples/power: Modify Power Main example

-Update power_main with new APP control functionality - blocking and unblocking
 power state transitions.
-We do not support triggering of PM_SLEEP from application side anymore.
-The application simply sleeps after blocking and unblocking state transition in
 an infinite loop.
-NOTE: This application does not test DVFS functionality yet.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

zhongnuo-tang committed:
os/board/rtl8730e/src/component/soc/amebad2: update runtime kernel verification

1. secureboot signing script need to modify when moving CA32 to XIP
2. runtime kernel verification need to modify according to add xip image
3. please use secure_boot_script_amebasmart_v04_official to sign

zhongnuo-tang committed:
os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_hp: include LCDC related code

1. include LCDC driver code
2. include MIPI-DSI interface
3. add AMEBASMART_MIPI config in Kconfig

zhongnuo-tang committed:
apps/examples/board_specific_demo: add MIPI demo example

1. example code for ST7701S to verify LCDC, MIPI driver code
2. contains st7701s_init() and related config
3. display maximum FPS (137) based on the configuration set in the example, on the top right corner of lcd screen
4. take note: there is no "maximum" fps, it should be based on the LCD resolution, in our example, the
LCD used has resolution 480*800, and pixel is RGB888 (24bits). More details can refer to ameba_mipi_show.c
5. LCD monitor in this example is ramless, which does not support changes of freq in realtime, and our LCDC
also does not support changes of freq in realtime. Hence, the fps will be fixed and correspond to the framerate (60)
set in the configuration in this example

Commit:
os/board/rtl8730e: Disable BT Classic in BT GAP lib

ewoodev committed:
build/tools/amebad/gnu_utility: Remove Flash operation in FTL init

When applying BLE hardfault commit(https://github.com/Samsung/TizenRT/commit/0f25fb2c02dc4964019bb736ca979af2e03306e9), the board is not booting well.

(Reset infinitely while booting)

To avoid this issue, this commit disables write protection in FTL init (remove flash operation)

Signed-off-by: jaeyong1-lee <jaeyong1.lee@samsung.com>
Signed-off-by: eunwoo.nam <eunwoo.nam@samsung.com>

ZhenBei-Sin committed:
os/board/rtl8730e: Move KM4 to XIP

Changes Note:
Bootloader support CA32 XIP image
Add prevention of double access to flash from KM4 and CA32
Utilize the KM4 SRAM for the KM4 heap

KM0 SRAM; KM4 PSRAM 92KB; ATF PSRAM 932KB;
Free up 7MB

Commit:
os/arch/arm/Kconfig: Remove ARM_HAVE_MPCORE for Cortex-A32
-ARM_HAVE_MPCORE is not needed for Cortex-A32
-Remove default select ARM_HAVE_MPCORE for Cortex-A32

jylee9613 committed:
framework/include/security : Increase secure storage data slot size

There is a slot that stores AP information, but the required size has increased while storing multi-AP information.
Since the number of slots is limited, it is more efficient to store the data by increasing the size of existing slots instead of dividing it into several slots.
Therefore, this commit increase data slot size from 256 to 4064.
Maximum size of a slot is 4KB,
but when data is stored in one slot, 32 bytes of tag is added.
Therefore, user can store 4064 bytes of data.

CookieDoughMixer committed:
os/arch/arm: allocate fiq stack based on CONFIG_ARMV7A_DECODEFIQ

fiq stack is allocated in all cases as of now. Hence, allocate fiq
stack only if CONFIG_ARMV7A_DECODEFIQ is enabled.

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

edwakuwaku committed:
os/arch/arm/src/amebasmart: Added PG and CG sleep flow, and frequency tuning for AP core

edwakuwaku committed:
os/board/rtl8730e: Added miscellaneous driver changes related to PM

edwakuwaku committed:
os/pm: PM driver change to integrate with application layer

edwakuwaku committed:
build/configs/rtl8730e: Place warm boot arch code to PSRAM

edwakuwaku committed:
os/board/rtl8730e: Release Loguart interrupt for AP

edwakuwaku committed:
build/tools/amebasmart: Update binary for KM0 core

edwakuwaku committed:
os/pm, os/arch/arm/src/amebasmart: Fix PM timer struct naming mismatch

edwakuwaku committed:
os/arch/arm/src/amebasmart, os/include: Refine DVFS API wrapper
- Update DVFS wrapper in pm.h
- Include PM header file in BSP level, and provide implementation

CookieDoughMixer committed:
kernel/task: refactor task_terminate function

make chages to add critial section only when needed. Also, move the
task cleanup logic to be executed after removing the task from the
task_list.

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

CookieDoughMixer committed:
configs/rtl8730e: adjust the stack of interrupt stacksize

on testing, the maximum stack used by interrupts is 232 bytes.
Hence, reduced the irq stack size to 1kb instead of 4kb

Signed-off-by: Abhishek Akkabathula <a.akkabathul@samsung.com>

ZhenBei-Sin committed:
os/board/rtl8730e: Delete unused folder-img3

Changes Note:
Delete the unused folder-img3 that violate the security verification.
No concern to delete it, this folder-img3 is an example on how to perform NonSecure access to Secure via NSC.

ZhenBei-Sin committed:
os/board/rtl8720e: Delete unused folders

Changes Note:
Delete the unused folder-img3 and example that violate the security verification.
No concern to delete it,
folder-img3 is an example on how to perform NonSecure access to Secure via NSC.
folder-example for the ssl client.

ZhenBei-Sin committed:
os/board/rtl8721csm: Delete unused cert and key defined in source file

Changes Note:
Delete the unused cert and key that violate the security verification.
No concern to delete it, as commented out as previous.

amogh-samsung committed:
os/pm: Correct alignments, spaces and descriptions.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
os/include/tinyara/pm: Add PM procfs path macros to tinyara header

-PM related procfs path macros should be available for use by any
 application to block/unblock state transition.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
apps/examples/power: Remove PM related procfs macros

-PM related procfs path macros are available for use by all
 applications in tinyara/pm/pm.h.
-Hence, include tinyara header file to use these path macros.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
os/pm: Modify procfs layer interface for PM control

-Procfs layer of PM should not be board dependant.
 tunefreq depends on CONFIG_DVFS, which in turn depends on board support
-Unify function for unlocking PM state transition. We will always support
 option of timer interrupt for waking up from PM_SLEEP, independant of board.
-We will not support the option of pushing the board to go to PM_SLEEP state
 from the app itself, hence remove enter_sleep_write function.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

amogh-samsung committed:
os: Add functionality for timer PM state locking

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

edwakuwaku commented on os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c:
```diff
@@ -51,7 +51,7 @@
 #include "timer_api.h"
 
 gtimer_t g_timer1;
```
How about having a g_timer2 here, specially for the 10mins hold for the BT?

edwakuwaku commented on os/pm/pm_procfs.c:
```diff
@@ -118,6 +118,7 @@ struct power_procfs_entry_s {
 
 /* Added to record timer countdown interrupt */
 struct pm_wakeup_timer_s g_timer_wakeup = { 0, 0 };
+struct pm_wakeup_timer_s g_timer_lock = { 0, 0 };
```
This might be redundant, we can expand the definition in pm_wakeup_timer_s, to handle more kind of utilizations.

edwakuwaku replied:
Instead of maintaining multiple pm_wakeup_timer_s (as the development going, we might have more timer utilization scenarios). Will it be better to expand definition of pm_wakeup_timer_s, to handle more kinds of scenario?

edwakuwaku commented on os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c:
```diff
@@ -123,6 +123,16 @@ void pg_timer_int_handler(void *Data)
 	pm_activity(PM_IDLE_DOMAIN, 9);
 }
 
+void pg_lock_timer_int_handler(void)
```
Instead of having 2 copies, we can add condition to check whether we are deploying timer1 or timer2, we can do that by expanding struct pm_timer_wakup_s
